Sources/SafeDICore/Models/TypeDescription.swift

```diff
@@ -240,6 +245,18 @@ public enum TypeDescription: Codable, Hashable, Comparable, Sendable {
             return self
         }
     }
+
+    /// A representation of this type that may be wrapped in a single element tuple to ensure cohesiveness of the type description.
+    private var wrappedIfAmbiguous: Self {
+        switch self {
+        case .void, .simple, .optional, .implicitlyUnwrappedOptional, .metatype, .nested, .array, .dictionary, .tuple, .unknown:
+            // These types contain no spaces, and are therefore unambiguous without being wrapped.
+            self
+        case .composition, .some, .any, .attributed, .closure:
+            // These types contain spaces and may be ambigous without being wrapped.
+            .tuple([.init(self)])
+        }
+    }
```

Comment:
I really like this. Simple.

Reply:
took me a few tries to whittle it down to something this contained.